app/deliver_grant_funding/routes/runner.py

```diff
@@ -0,0 +1,94 @@
+from uuid import UUID
+
+from flask import redirect, render_template, request, session, url_for
+from flask.typing import ResponseReturnValue
+
+from app.common.auth.decorators import has_grant_role
+from app.common.collections.runner import DGFFormRunner
+from app.common.data import interfaces
+from app.common.data.types import FormRunnerState, RoleEnum
+from app.deliver_grant_funding.routes import deliver_grant_funding_blueprint
+from app.extensions import auto_commit_after_request
+
+
+@deliver_grant_funding_blueprint.route(
+    "/grant/<uuid:grant_id>/submissions/<uuid:submission_id>", methods=["GET", "POST"]
+)
+@auto_commit_after_request
+@has_grant_role(RoleEnum.MEMBER)
+def submission_tasklist(grant_id: UUID, submission_id: UUID) -> ResponseReturnValue:
+    source = request.args.get("source")
+    runner = DGFFormRunner.load(submission_id=submission_id, source=FormRunnerState(source) if source else None)
```

Comment:
These pages will also need to start validating that the user has access to the given submission. We've skipped this for the developer pages, but for the real pages I think we can't ignore this anymore.

For test submissions this means that user A can't access user B's test submission. For live submissions, it will mean that anyone in Org A can't access Org B's submissions.

We'll need to think about the best place to put this - probably in the FormRunner as part of retrieving the submission via `get_submission` - might be that `get_submission` _requires_ passing in a user and that's fed through to the SQL queries to do the checks there.